libsolidity/codegen/Compiler.h

```diff
@@ -34,9 +35,11 @@ namespace solidity {
 class Compiler
 {
 public:
-	explicit Compiler(langutil::EVMVersion _evmVersion = langutil::EVMVersion{}, bool _optimize = false, unsigned _runs = 200):
-		m_optimize(_optimize),
-		m_optimizeRuns(_runs),
+	explicit Compiler(
+		langutil::EVMVersion _evmVersion = langutil::EVMVersion{},
+		OptimiserSettings _optimiserSettings = OptimiserSettings::minimal()
```

Comment:
Do we actually need defaults here? Is this called without explicit arguments anywhere?

Reply:
Should I also remove the default for the version?

Reply:
I'd say nothing below `CompilerStack` should have a default.